File: day11/password.rb
def valid?(str)
  return false if str.match(/i|l|o/) #any pw with these letters is invalid
  match = str.match(/(\w)\1/)
  if match
    double = str.index(match[0]) #find where double letter starts
    temp = str.sub(match[0], '--') #replace double letter with non-alphanumeric chars
    return false unless !temp.match(match[0]) and temp.match(/(\w)\1/) #bail early unless
    #candidate doesn't contain the first double letter and does contain a different double letter
  else
    return false  #no double letters at all
  end
  if str.match(/abc|bcd|cde|def|efg|fgh|pqr|qrs|rst|stu|tuv|uvw|vwx|wxy|xyz/)
    return true
  else
    return false
  end
end

password = 'vzbxkghb'
valid = false
until valid
  password.succ!
  valid = valid?(password)
end

puts password

File: day11/password.py
from re import findall,search

def get_next(letters):
    i = len(letters)-1
    letters = list(letters)
    while i >= 0:
        if letters[i] == 'z':
            letters[i] = 'a'
            i -= 1
        else:
            if letters[i] in [ 'h', 'k', 'n' ]:
                inc_amount = 2    
            else: 
                inc_amount = 1
            letters[i] = chr(ord(letters[i])+inc_amount)
            return ''.join(letters)

def validate_password(pw):
    if len(findall(r'(\w)\1',pw)) < 2:
        return False
    if not search(r'abc|bcd|cde|def|efg|fgh|pqr|qrs|rst|stu|tuv|uvw|vwx|wxy|xyz',pw):
        return False
    return True

if __name__ == '__main__':
    letters = 'vzbxkghb'
    is_valid = False
    while not is_valid:
        letters = get_next(letters)
        is_valid = validate_password(letters)
    print(letters)
    is_valid = False
    while not is_valid:
        letters = get_next(letters)
        is_valid = validate_password(letters)
    print(letters)
   